### DISCUSSION TITLE
I am not able to generate certificate. Getting error. "The delegated name server is not correct."

### DISCUSSION THREAD

**Discussion by mbhshah** [polymind-inc/acmebot#670] [General]
I have used below steps to setup environment.

1) I have deployed function app using azure public. And enabled app service authentication.

2) I have created new azure dns zone named wmdevappsportal.com and provided dns zone contributor access to function app from dns zone resource. But I am getting error as shown below screen shot. Can anyone please help.

![image](https://github.com/shibayan/appservice-acmebot/assets/149039585/d9bf3d4d-8cb5-44ba-8abd-f225f13a6c5b)

![image](https://github.com/shibayan/appservice-acmebot/assets/149039585/3b180e26-fe94-4fff-b260-986a6c7abe66)

![image](https://github.com/shibayan/keyvault-acmebot/assets/149039585/3f376e7b-28fd-461e-a50b-d2d26a488f28)

**Comment by shibayan** [comment]
Have you delegated to Azure DNS name servers at your domain registrar? Creating an Azure DNS resource is not enough, you must change the nameservers at your domain registrar.

https://learn.microsoft.com/en-us/azure/dns/dns-domain-delegation

**Reply by mbhshah** [reply]
I have not purchased this wmdevappsportal.com domain from any registrar like GoDaddy or any other. Do I have to purchase it?

**Reply by shibayan** [reply]
You do not need to purchase, but you will need to delegate your name servers to Azure DNS. Please consult your administrator.

**Reply by mbhshah** [reply]
Thank you for the reply.

I was going through below link. But I can't find below highlighted steps. Can you please guide?

https://learn.microsoft.com/en-us/azure/dns/dns-delegate-domain-azure-dns

![image](https://github.com/shibayan/keyvault-acmebot/assets/149039585/e97c6458-9085-40da-b6a8-70d4dd87b7a7)

**Reply by shibayan** [reply]
We cannot support changing the name servers for a domain because it is not related to Azure and depends on how the domain was acquired.
